internal/cmd/finding/finding.go

```diff
@@ -86,50 +91,130 @@ func newWithOptions(opts *options) *cobra.Command {
+	} else {
+		log.Infof(messaging.UsageWarning())
+	}
+
+	localFindings, err := finding.LocalFindings(cmd.opts.ProjectDir, errorDetails)
+	if err != nil {
+		return err
+	}
+
+	// store remote findings in a slice of finding.Finding so that we can search
+	// them individually later. These won't be stored on disk.
+	var remoteFindings []*finding.Finding
+	for i := range remoteAPIFindings.Findings {
+		// we access the element via index to avoid copying the struct
+		rf := remoteAPIFindings.Findings[i]
+
+		timeStamp, err := time.Parse(time.RFC3339, rf.Timestamp)
+		if err != nil {
+			return errors.Wrapf(err, "Could not parse timestamp %s", rf.Timestamp)
+		}
+		remoteFindings = append(remoteFindings, &finding.Finding{
+			Origin:             "CI Sense",
+			Name:               strings.TrimPrefix(rf.Name, fmt.Sprintf("projects/%s/findings/", cmd.opts.Project)),
+			Type:               finding.ErrorType(rf.ErrorReport.Type),
+			InputData:          rf.ErrorReport.InputData,
+			Logs:               rf.ErrorReport.Logs,
+			Details:            rf.ErrorReport.Details,
+			HumanReadableInput: string(rf.ErrorReport.InputData),
+			MoreDetails:        rf.ErrorReport.MoreDetails,
+			Tag:                rf.ErrorReport.Tag,
+			CreatedAt:          timeStamp,
+			FuzzTest:           rf.FuzzTargetDisplayName,
+			StackTrace: []*stacktrace.StackFrame{
+				{
+					Function:   rf.ErrorReport.DebuggingInfo.BreakPoints[0].Function,
+					SourceFile: rf.ErrorReport.DebuggingInfo.BreakPoints[0].SourceFilePath,
+					Line:       rf.ErrorReport.DebuggingInfo.BreakPoints[0].Location.Line,
+					Column:     rf.ErrorReport.DebuggingInfo.BreakPoints[0].Location.Column,
+				},
+			},
+		})
+	}
+
 	if len(args) == 0 {
 		// If called without arguments, `cifuzz findings` lists short
 		// descriptions of all findings
-		findings, err := finding.ListFindings(cmd.opts.ProjectDir, errorDetails)
-		if err != nil {
-			return err
-		}
+		allFindings := append(localFindings, remoteFindings...)
 
 		if cmd.opts.PrintJSON {
-			s, err := stringutil.ToJSONString(findings)
+			s, err := stringutil.ToJSONString(allFindings)
 			if err != nil {
 				return err
 			}
 			_, _ = fmt.Fprintln(cmd.OutOrStdout(), s)
 			return nil
 		}
 
-		if len(findings) == 0 {
+		if len(allFindings) == 0 {
 			log.Print("This project doesn't have any findings yet")
 			return nil
 		}
 
 		w := tabwriter.NewWriter(cmd.OutOrStdout(), 0, 0, 1, ' ', 0)
 
 		data := [][]string{
-			{"Severity", "Name", "Description", "Fuzz Test", "Location"},
+			{"Origin", "Severity", "Name", "Description", "Fuzz Test", "Location"},
 		}
 
-		for _, f := range findings {
+		for _, f := range allFindings {
 			score := "n/a"
 			locationInfo := "n/a"
 			// add location (file, function, line) if available
-			if len(f.ShortDescriptionColumns()) > 1 {
+			if f.StackTrace != nil && len(f.StackTrace) > 0 {
+				locationInfo = fmt.Sprintf("%s:%s:%d", f.StackTrace[0].SourceFile, f.StackTrace[0].Function, f.StackTrace[0].Line)
```

Comment:
This produces output like `src/explore_me.cpp:exploreMe:13`. I would prefer omitting the function name here to produce the more common format `$FILE:$LINE` which is understood by existing tools, for example the "Go to file" action in JetBrains IDEs will bring you directly to the line in that file if you copy paste the string.

```suggestion
				locationInfo = fmt.Sprintf("%s:%d", f.StackTrace[0].SourceFile, f.StackTrace[0].Line)
```

Reply:
What I would like even better is to extract a method `finding.SourceLocation` which can be used here and in the `finding.ShortDescriptionColumns` method.

Reply:
Thanks! I added 8b50eea that should do this.